src/structs.py

```diff
@@ -710,6 +730,25 @@ def pddl_str(self) -> str:
                             self.side_predicates)
         return op.pddl_str()
 
+    def pretty_str(self, name_map: Dict[str, str]) -> str:
+        """Display the NSRT in a nice human-readable format, given a mapping to
+        new predicate names for any invented predicates."""
+        out = ""
+        out += f"{self.name}:\n\tParameters: {self.parameters}"
+        for name, atoms in [("Preconditions", self.preconditions),
+                            ("Add Effects", self.add_effects),
+                            ("Delete Effects", self.delete_effects)]:
+            out += f"\n\t{name}:"
+            for atom in atoms:
+                pretty_pred = atom.predicate.pretty_str()[1]
+                new_name = (name_map[pretty_pred] if pretty_pred in name_map
+                            else str(atom.predicate))
+                var_str = ", ".join(map(str, atom.variables))
+                out += f"\n\t\t{new_name}({var_str})"
+        option_var_strs = [str(v) for v in self.option_vars]
+        out += f"\n\tOption Spec: ({self.option.name}, {option_var_strs})"
```

Comment:
for paper, we'll want to switch to Controller, but keeping this consistent with codebase here